### PR TITLE
Learning curves

### PR DESCRIPTION
**Pull request by juhoinkinen**
It would be useful to be able to easily estimate "what is enough training data", in which [learning curve](https://scikit-learn.org/stable/modules/learning_curve.html#learning-curve) could help. 

In this draft there is a command for Annif CLI to generate `.tsv` log files with metrics evaluated on both (fixed size) test set and training set, when training set size is increased. An example of a test-set log: 

```
Documents in training	Precision (doc avg)	Recall (doc avg)	F1 score (doc avg)	NDCG	NDCG@5	NDCG@10	Documents evaluated
295	0.0751908396946565	0.17577547558463588	0.10451335802114523	0.16634240232662204	0.14813915760140667	0.16634240232662204	262
591	0.09274809160305343	0.21440082394280865	0.1285088476778974	0.21819156188397468	0.19707598246780478	0.21819156188397468	262
886	0.10152671755725191	0.23339997576638796	0.1404417370063271	0.22667717388613576	0.20125209246306985	0.22667717388613576	262
1182	0.10458015267175572	0.2391251666060826	0.14434472333297751	0.23184537405108485	0.20362806264014807	0.2319126490100198	262
1477	0.10763358778625955	0.24627105295044224	0.14861153760883158	0.24145462747026986	0.21190518206988485	0.2415092216082432	262
1772	0.11259541984732824	0.2575730037562099	0.1555463842674629	0.2561306252425605	0.23112553419393153	0.2561306252425605	262
2068	0.11641221374045801	0.2661062643887071	0.1607701531643346	0.2652253917961684	0.2398870180508398	0.2652253917961684	262
2363	0.1183206106870229	0.2705864534108809	0.16345179834148946	0.27009438372371863	0.24399389524829707	0.27009438372371863	262
```

### CODE REVIEW

**Comment by juhoinkinen** [comment]
Closing this unmerged because the learning curves can now be produced by a shell script with training and evaluation steps using the `--docs-limit` option (#465).
